versioned/transfer/src/main/java/org/projectnessie/versioned/transfer/ExportContents.java

```diff
@@ -0,0 +1,170 @@
+import org.projectnessie.nessie.relocated.protobuf.ByteString;
+import org.projectnessie.versioned.CommitMetaSerializer;
+import org.projectnessie.versioned.GetNamedRefsParams;
+import org.projectnessie.versioned.KeyEntry;
+import org.projectnessie.versioned.ReferenceInfo;
+import org.projectnessie.versioned.ReferenceNotFoundException;
+import org.projectnessie.versioned.VersionStore;
+import org.projectnessie.versioned.paging.PaginationIterator;
+import org.projectnessie.versioned.persist.adapter.spi.AbstractDatabaseAdapter;
+import org.projectnessie.versioned.storage.common.objtypes.Hashes;
+import org.projectnessie.versioned.store.DefaultStoreWorker;
+import org.projectnessie.versioned.transfer.files.ExportFileSupplier;
+import org.projectnessie.versioned.transfer.serialize.TransferTypes.Commit;
+import org.projectnessie.versioned.transfer.serialize.TransferTypes.ExportVersion;
+import org.projectnessie.versioned.transfer.serialize.TransferTypes.HeadsAndForks;
+import org.projectnessie.versioned.transfer.serialize.TransferTypes.NamedReference;
+import org.projectnessie.versioned.transfer.serialize.TransferTypes.Operation;
+import org.projectnessie.versioned.transfer.serialize.TransferTypes.OperationType;
+import org.projectnessie.versioned.transfer.serialize.TransferTypes.RefType;
+
+/**
+ * Creates export artifacts by generating artificial commits from the latest content object values
+ * on one particular branch.
+ */
+final class ExportContents extends ExportCommon {
+  private final VersionStore store;
+
+  private ByteString lastCommitId = AbstractDatabaseAdapter.NO_ANCESTOR.asBytes();
+
+  ExportContents(ExportFileSupplier exportFiles, NessieExporter exporter) {
+    super(exportFiles, exporter);
+    store = exporter.versionStore();
+  }
+
+  @Override
+  ExportVersion getExportVersion() {
+    return ExportVersion.V1;
+  }
+
+  @Override
+  void writeRepositoryDescription() {}
+
+  private <T> List<T> take(int n, Iterator<T> it) {
+    List<T> result = new ArrayList<>(n);
+    while (n-- > 0 && it.hasNext()) {
+      result.add(it.next());
+    }
+    return result;
+  }
+
+  @SuppressWarnings("UnstableApiUsage")
+  @Override
+  HeadsAndForks exportCommits(ExportContext exportContext) {
+    ReferenceInfo<CommitMeta> ref;
+    try {
+      ref = store.getNamedRef(exporter.contentsFromBranch(), GetNamedRefsParams.DEFAULT);
+    } catch (ReferenceNotFoundException e) {
+      throw new RuntimeException(e);
+    }
+
+    long startMicors = TimeUnit.MILLISECONDS.toMicros(currentTimestampMillis());
+
+    long seq = 0;
+    try (PaginationIterator<KeyEntry> entries = store.getKeys(ref.getNamedRef(), null, false)) {
+      while (true) {
+        List<KeyEntry> batch = take(exporter.contentsBatchSize(), entries);
+        if (batch.isEmpty()) {
+          break;
+        }
+
+        ByteString meta =
+            CommitMetaSerializer.METADATA_SERIALIZER.toBytes(
+                CommitMeta.fromMessage("Export " + seq));
```

Comment:
Maybe `Single branch export from '<branch>', part #<part-num>` - and `... last part` for the last one. And `seq + 1`?

Reply:
Updated the message... The `last part` looks like an overkill to me... skipped that ;)